File: server/src/main/java/br/app/pregsrateio/usuario/data/UsuarioRepository.java
package br.app.pregsrateio.usuario.data;

import org.bson.types.ObjectId;
import org.springframework.data.mongodb.repository.MongoRepository;

import java.util.Optional;

public interface UsuarioRepository extends MongoRepository<Usuario, ObjectId> {
    Optional<Usuario> findByAuthSub(String authSub);
}

File: server/src/main/kotlin/br/dev/brunorsch/pregsrateio/usuario/model/UsuarioRepository.kt
package br.dev.brunorsch.pregsrateio.usuario.model

import io.micronaut.data.mongodb.annotation.MongoRepository
import io.micronaut.data.repository.CrudRepository
import org.bson.types.ObjectId

@MongoRepository
interface UsuarioRepository : CrudRepository<Usuario, ObjectId> {
    fun findByAuthSub(authSub: String): Usuario?
}